[Inductor] Modify persistent+TMA template for Triton mm and admm to use new TMA API

NikhilAPatel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #151775
* __->__ #151774

Summary:
This PR modifies the Triton template for persisten+TMA mm and admm to use the new functional API for TMA introduced here: https://github.com/triton-lang/triton/pull/6248/

This also involves setting a global Triton allocator function to be called at kernel launch for any kernels that require additional global memory workspace. This is done in triton_heuristics.py directly before kernels are launched.

Test Plan:
contbuild & OSS CI

Reviewers: paulzhan

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

NikhilAPatel committed:
[Inductor] Modify persistent+TMA template for Triton mm and admm to use new TMA API

Summary:
This PR modifies the Triton template for persisten+TMA mm and admm to use the new functional API for TMA introduced here: https://github.com/triton-lang/triton/pull/6248/

This also involves setting a global Triton allocator function to be called at kernel launch for any kernels that require additional global memory workspace. This is done in triton_heuristics.py directly before kernels are launched.

Test Plan:
contbuild & OSS CI

Reviewers: paulzhan

[ghstack-poisoned]

PaulZhang12 reviewed (approved):
Nice! LGTM, is there a local script you used for testing? Also, we should be able to merge this now given Triton 3.3 is out?

NikhilAPatel commented:
> Nice! LGTM, is there a local script you used for testing? Also, we should be able to merge this now given Triton 3.3 is out?

I do have a local script but its pretty bare bones. The majority of the testing was done using TritonBench and test/inductor/test_max_autotune.py

Yeah we should be able to merge it now, I'll rebase it and see if it passes all the tests.

NikhilAPatel commented:
@pytorchbot merge -r

NikhilAPatel committed:
Update

[ghstack-poisoned]

NikhilAPatel committed:
Update on "[Inductor] Modify persistent+TMA template for Triton mm and admm to use new TMA API"


Summary:
This PR modifies the Triton template for persisten+TMA mm and admm to use the new functional API for TMA introduced here: https://github.com/triton-lang/triton/pull/6248/

This also involves setting a global Triton allocator function to be called at kernel launch for any kernels that require additional global memory workspace. This is done in triton_heuristics.py directly before kernels are launched.

Test Plan:
contbuild & OSS CI

Reviewers: paulzhan

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]
